nilearn/surface/surface.py

```diff
@@ -1753,3 +1779,167 @@ def from_volume(
         data = PolyData(left=texture_left, right=texture_right)
 
         return cls(mesh=mesh, data=data)
+
+
+def check_surface_data_ndims(img, dim, var_name="img"):
+    """Check if the data of a SurfaceImage is of a given dimension,
+    raise error if not.
+
+    Parameters
+    ----------
+    img : :obj:`~nilearn.surface.SurfaceImage`
+        SurfaceImage to check.
+
+    dim : int
+        Dimensions the data should have.
+
+    var_name : str, optional
+        Name of the variable to include in the error message.
+
+    Returns
+    -------
+    raise ValueError if the data of the SurfaceImage is not of the given
+    dimension.
+    """
+    n_dim_left = img.data.parts["left"].ndim
+    n_dim_right = img.data.parts["right"].ndim
+    if not all(x == dim for x in [n_dim_left, n_dim_right]):
+        raise ValueError(
+            f"Data for each hemisphere of {var_name} should be {dim}D, "
+            f"but found {n_dim_left}D for left and {n_dim_right}D for right."
+        )
+
+
+def get_min_max_surface_image(img):
+    """Get min and max across hemisphere for a SurfaceImage.
+
+    Parameters
+    ----------
+    img : SurfaceImage
+
+    Returns
+    -------
+    vmin : float
+
+    vmax : float
+    """
+    vmin = min(min(x.ravel()) for x in img.data.parts.values())
+    vmax = max(max(x.ravel()) for x in img.data.parts.values())
+    return vmin, vmax
+
+
+def concat_extract_surface_data_parts(img):
```

Comment:
In the same vein, how about renaming this to `get_data`? One caveat might be that we are doing concatenation across hemispheres but that could be mentioned in the docstring.
https://github.com/nilearn/nilearn/blob/04be71faa15194e41e9455ca87d3b4427580ff42/nilearn/image/image.py#L46
<img width="845" alt="Screenshot 2024-12-17 at 12 07 46 PM" src="https://github.com/user-attachments/assets/7069cdfb-ba71-4ca2-8894-d27f44de243c" />

Reply:
sounds fair indeed